### DISCUSSION TITLE
LocalNotificationCenter.Current.RequestNotificationPermission Throws NullPointerException

### DISCUSSION THREAD

**Discussion by Srki94** [thudugala/Plugin.LocalNotification#463] [Q&A] [answered]
Hi,

I'm using this plugin to notify users daily, using Xamarin Forms Android app. 
If  I set Alarm and Reminders, plus Notifications permissions prior to starting the app, the App works.
However, my users need to do this often with new updates etc. because they can't login otherwise as my login is wrapped in Try Catch block. 

Isn't RequestNotificationPermission() supposed to request the permission? 
I get the mentioned error.

Here's the chunk of code :

` if (await LocalNotificationCenter.Current.AreNotificationsEnabled() == false)
 {
     if (await LocalNotificationCenter.Current.RequestNotificationPermission())
     {
         LocalNotificationCenter.Current.CancelAll();
         RegisterReminders();
     }
 }
 else
 {
     LocalNotificationCenter.Current.CancelAll();
     RegisterReminders();
 }`
 
 Any help would be appreciated, thanks!
 
 Here's the full trace : 

>  {Java.Lang.NullPointerException: Attempt to invoke virtual method 'void android.app.Activity.requestPermissions(java.lang.String[], int)' on a null object reference
>   at Java.Interop.JniEnvironment+StaticMethods.CallStaticVoidMethod (Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00068] in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/obj/Release/JniEnvironment.g.cs:13250 
>   at Java.Interop.JniPeerMembers+JniStaticMethods.InvokeVoidMethod (System.String encodedMember, Java.Interop.JniArgumentValue* parameters) [0x00008] in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop/Java.Interop/JniPeerMembers.JniStaticMethods.cs:97 
>   at AndroidX.Core.App.ActivityCompat.RequestPermissions (Android.App.Activity activity, System.String[] permissions, System.Int32 requestCode) [0x00057] in /Users/runner/work/1/s/generated/androidx.core.core/obj/Release/monoandroid12.0/generated/src/AndroidX.Core.App.ActivityCompat.cs:555 
>   at Plugin.LocalNotification.LocalNotificationCenter.RequestNotificationPermissionAsync (Plugin.LocalNotification.NotificationPermission permission) [0x000b3] 
> 
> java.lang.NullPointerException: Attempt to invoke virtual method 'void android.app.Activity.requestPermissions(java.lang.String[], int)' on a null object reference
> 	at androidx.core.app.ActivityCompat$Api23Impl.requestPermissions(ActivityCompat.java:893)
> 	at androidx.core.app.ActivityCompat.requestPermissions(ActivityCompat.java:557)
> 	at mono.java.lang.RunnableImplementor.n_run(Native Method)
> 	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:31)
> 	at android.os.Handler.handleCallback(Handler.java:958)
> 	at android.os.Handler.dispatchMessage(Handler.java:99)
> 	at android.os.Looper.loopOnce(Looper.java:230)
> 	at android.os.Looper.loop(Looper.java:319)
> 	at android.app.ActivityThread.main(ActivityThread.java:8893)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
> }

**Comment by thudugala** [comment, marked as answer]
@Srki94 https://github.com/thudugala/Plugin.LocalNotification/wiki/2.-Usage-10.0.0-Xamarin.Forms

try this

![image](https://github.com/thudugala/Plugin.LocalNotification/assets/4112014/cf6b4be1-b3fc-4889-a298-923395e97a56)
